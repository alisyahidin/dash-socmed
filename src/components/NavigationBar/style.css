.my-navbar {
  box-shadow: 0px 1px 1px rgba(0, 0, 0, 0.1);
}

.my-btn {
  margin: 0.2rem 0;
}

#menu:hover {
  cursor: pointer !important;
}

.nav-link__menu {
  display: flex !important;
  align-items: center;
}

.nav-link.active {
  color: rgb(79, 145, 228) !important;
}