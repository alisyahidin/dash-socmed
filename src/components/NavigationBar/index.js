import React, { Component } from 'react'
import { Link, withRouter } from 'react-router-dom'
import classNames from 'classnames'
import {
  Container,
  Nav,
  Navbar,
  Dropdown
} from 'react-bootstrap'
import {
  MdHome,
  MdGroup,
  MdKeyboardArrowDown
} from "react-icons/md"
import './style.scss'

class ProfileMenu extends Component {
  render() {
    const { children, onClick } = this.props

    return (
      <span
        id="menu"
        onClick={onClick}
        className="nav-link mr-sm-2"
      >
        {children}
      </span>
    )
  }
}

const ProfileMenuItem = ({onClick, children}) => (
  <Link to="/profile" onClick={onClick} className="dropdown-item">
    {children}
  </Link>
)

class NavigationBar extends Component {
  render() {
    const { pathname } = this.props.location
    return (
      <Navbar bg="white" variant="light" className="my-navbar">
        <Container>
          <Nav className="mr-auto">
            <Link
              to="/"
              className={classNames(
                "nav-link nav-link__menu",
                { active: pathname === '/' }
              )}
            >
              <MdHome className="mr-sm-1" /> Home
            </Link>
            <Link
              to="/users"
              className={classNames(
                "nav-link nav-link__menu",
                { active: pathname === '/users' }
              )}
            >
              <MdGroup className="mr-sm-1" /> Users
            </Link>
          </Nav>
          <Nav className="ml-auto">
            <Dropdown>
              <Dropdown.Toggle as={ProfileMenu} id="profile-menu">
                Ali Syahidin <MdKeyboardArrowDown />
              </Dropdown.Toggle>

              <Dropdown.Menu>
                <Dropdown.Item as={ProfileMenuItem}>Profile</Dropdown.Item>
                <Dropdown.Item>Logout</Dropdown.Item>
              </Dropdown.Menu>
            </Dropdown>
          </Nav>
        </Container>
      </Navbar>
    )
  }
}

export default withRouter(NavigationBar)