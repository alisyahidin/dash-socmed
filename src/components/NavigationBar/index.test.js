import React from 'react'
import { shallow } from 'enzyme'
import NavigationBar from './index'

it('render <NavigationBar /> correctly', () => {
  const component = shallow(<NavigationBar />)

  expect(component).toMatchSnapshot()
})