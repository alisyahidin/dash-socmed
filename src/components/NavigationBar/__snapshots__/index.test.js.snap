// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <NavigationBar /> correctly 1`] = `
<Route>
  <Component />
</Route>
`;
